plugin/trino-phoenix/src/main/java/io/trino/plugin/phoenix/PhoenixClient.java

```diff
@@ -280,6 +283,33 @@ public PreparedStatement prepareStatement(
         return new QueryBuilder(this).prepareStatement(session, connection, preparedQuery);
     }
 
+    @Override
+    public boolean supportsTopN(ConnectorSession session, JdbcTableHandle handle, List<JdbcSortItem> sortOrder)
+    {
+        return true;
+    }
+
+    @Override
+    protected Optional<TopNFunction> topNFunction()
+    {
+        return Optional.of((query, sortItems, limit) -> {
+            // TODO: Remove when this is improved in Phoenix
+            // Phoenix performs a client side merge sort.
+            // For large sets this is slower than passing
+            // the unsorted data to Trino and sort it there.
```

Comment:
I would expect this comment to talk about memory implications of changing (or not having) the limit, or a link to Phoenix issue describing that.

Also, if the Trino-side sorting is faster, I am not sure the TODO is actionable. If it is not, please just remove the "TODO" and please retain the rest of the info.